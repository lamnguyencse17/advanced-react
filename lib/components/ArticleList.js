import React, { Component } from 'react';
import Article from "./Article";

export default class ArticleList extends Component {
    render() {
        return (
            <div>
                {Object.values(this.props.articles).map(article =>
                <Article 
                    key={article.id}
                    article={article}
                    store={this.props.store}
                />
                )}
            </div>
        );
    }
}
