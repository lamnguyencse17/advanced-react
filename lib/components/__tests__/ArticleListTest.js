import React from 'react';
import ArticleList from '../ArticleList';
import DataApi from '@lib/DataApi';
import {data} from '@lib/testData';

const store = new DataApi(data);

import renderer from 'react-test-renderer';

describe('ArticleList', () => {
    const testProps = {
        articles: store.getState().articles,
        store: store
    };
    
    it('renders correctly', () => {
        const tree = renderer.create(
            <ArticleList {...testProps}/>
        ).toJSON();
        
        expect(tree.children.length).toBe(5);
        expect(tree).toMatchSnapshot();
    });
});
