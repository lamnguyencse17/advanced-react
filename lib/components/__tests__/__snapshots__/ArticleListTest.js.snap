// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "bordeerBottomStyle": 10,
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWegiht": "bold",
        }
      }
    >
      Introducing React's Error Code System
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href="https://twitter.com/keyanzhang"
      >
        Keyan
         
        Zhang
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.
    </div>
  </div>
  <div
    style={
      Object {
        "bordeerBottomStyle": 10,
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWegiht": "bold",
        }
      }
    >
      Mixins Considered Harmful
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Wed Jul 13 2016 00:00:00 GMT+0000 (UTC)
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href="https://twitter.com/dan_abramov"
      >
        Dan
         
        Abramov
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      “How do I share the code between several components?” is one of the first questions that people ask when they learn React. Our answer has always been to use component composition for code reuse. You can define a component and use it in several other components. 
It is not always obvious how a certain pattern can be solved with composition. React is influenced by functional programming but it came into the field that was dominated by object-oriented libraries. It was hard for engineers both inside and outside of Facebook to give up on the patterns they were used to.
    </div>
  </div>
  <div
    style={
      Object {
        "bordeerBottomStyle": 10,
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWegiht": "bold",
        }
      }
    >
      Create Apps with No Configuration
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href="https://twitter.com/dan_abramov"
      >
        Dan
         
        Abramov
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      Create React App is a new officially supported way to create single-page React applications. It offers a modern build setup with no configuration. 

Getting Starte 
Installation 
First, install the global package:
    </div>
  </div>
  <div
    style={
      Object {
        "bordeerBottomStyle": 10,
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWegiht": "bold",
        }
      }
    >
      Relay: State of the State
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Fri Aug 05 2016 00:00:00 GMT+0000 (UTC)
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href="https://twitter.com/en_JS"
      >
        Joseph
         
        Savona
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      This month marks a year since we released Relay and we'd like to share an update on the project and what's next. 
A Year In Review 
A year after launch, we're incredibly excited to see an active community forming around Relay and that companies such as Twitter are using Relay in production:
    </div>
  </div>
  <div
    style={
      Object {
        "bordeerBottomStyle": 10,
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWegiht": "bold",
        }
      }
    >
      React v15.5.0
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Fri Apr 07 2017 00:00:00 GMT+0000 (UTC)
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href="https://twitter.com/acdlite"
      >
        Andrew
         
        Clark
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      It's been exactly one year since the last breaking change to React. Our next major release, React 16, will include some exciting improvements, including a complete rewrite of React's internals. We take stability seriously, and are committed to bringing those improvements to all of our users with minimal effort.
 To that end, today we're releasing React 15.5.0.
    </div>
  </div>
</div>
`;
